x-pack/plugins/ml/public/application/timeseriesexplorer/components/timeseries_chart/timeseries_chart.js

```diff
@@ -1798,7 +1799,15 @@ class TimeseriesChartIntl extends Component {
 }
 
 export const TimeseriesChart = (props) => {
-  const annotationProp = useObservable(annotation$);
+  const annotationProp = useObservable(
+    annotation$.pipe(
```

Comment:
seems like this code is a copy-paste of https://github.com/elastic/kibana/blob/b1edce8050cc209053d094f1788b7de657123ae2/x-pack/plugins/ml/public/application/components/annotations/annotation_flyout/index.tsx#L407

I think it's worth using one observable service for both cases. Instead of the statically exposed subject (it's a bad practice), we should inject an observable instance into components instead. 
So it could be a service, let's call it `AnnotationUpdatesServcie` with 2 methods, `updates$` that expose subject as observable (simply with `.asObservable()`), using the pipe with `distinctUntilChanged `, and `setValue` method that calls `next` under the hood

Reply:
Created a `AnnotationUpdatesService` and `MlAnnotationComponent` to wrap round `TimeSeriesChart` and `AnnotationFlyout` here https://github.com/elastic/kibana/pull/83216/commits/b5166e64604e87ebfad5e3263b1ddf4dc1ef3788